plugins/rest-plugin/src/main/java/ru/sbtqa/tag/api/EndpointEntryReflection.java

```diff
@@ -0,0 +1,159 @@
+                if (annotation instanceof Body && ((Body) annotation).name().equals(name)
+                        || annotation instanceof Query && ((Query) annotation).name().equals(name)
+                        || annotation instanceof Header && ((Header) annotation).name().equals(name)
+                        && value != null && !value.isEmpty()) {
+                    set(endpoint, field, value);
+                    return;
+                }
+            }
+        }
+
+        throw new RestPluginException("There is no '" + name + "' parameter in '" + endpoint.getClass().getAnnotation(ru.sbtqa.tag.api.annotation.Endpoint.class).title() + "' endpoint.");
+    }
+
+    public String getBody(String template) {
+        String templatePath = template.isEmpty() ? endpoint.getClass().getSimpleName() : template;
+        String body = loadTemplateFile(templatePath);
+
+        return replaceParameters(body);
+    }
+
+    private String loadTemplateFile(String templatePath) {
+        try {
+            return IOUtils
+                    .toString(getClass().getClassLoader().getResourceAsStream(templatePath), PROPERTIES.getTemplateEncoding())
+                    .replace(BOM, "");
+        } catch (NullPointerException ex) {
+            throw new RestPluginException("Can't find template file by path " + templatePath, ex);
+        } catch (IOException ex) {
+            throw new RestPluginException("Template file '" + templatePath + "' is not available", ex);
+        }
+    }
+
+    private String replaceParameters(String body) {
+        for (Map.Entry<String, Object> parameter : getParameters(ParameterType.BODY).entrySet()) {
+            if (parameter.getValue() instanceof String) {
+                String value = (String) parameter.getValue();
+                body = body.replaceAll("%" + parameter.getKey(), value);
+            } else {
+                LOG.debug("Failed to substitute not String field to body template");
+            }
+        }
+
+        return body;
+    }
+
+
+    public void validate(String title, Object... params) {
+        Method[] methods = endpoint.getClass().getMethods();
+        for (Method method : methods) {
+            if (null != method.getAnnotation(Validation.class)
+                    && method.getAnnotation(Validation.class).title().equals(title)) {
+                try {
+                    method.invoke(endpoint, params);
+                } catch (InvocationTargetException | IllegalAccessException | IllegalArgumentException e) {
+                    throw new RestPluginException("Failed to invoke method", e);
+                }
+                return;
+            }
+        }
+
+        throw new RestPluginException("There is no '" + title + "' validation rule in '" + endpoint.getClass().getAnnotation(ru.sbtqa.tag.api.annotation.Endpoint.class).title() + "' endpoint.");
+    }
+
+    public Map<String, Object> getParameters(ParameterType type) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=81&issues=AWU2KVJuW2AOP0CkD5Fx&open=AWU2KVJuW2AOP0CkD5Fx)

[ISSUE_KEY]: AWU2KVJuW2AOP0CkD5Fx